language/move-prover/tests/sources/functional/verify_vector.move

```diff
@@ -145,16 +146,33 @@ module VerifyVector {
         ensures forall i in 0..len(v): old(v[i]) == v[len(v)-1-i];
     }
 
-    // Moves all of the elements of the `other` vector into the `lhs` vector.
-    fun verify_append<Element>(lhs: &mut vector<Element>, other: vector<Element>) {
+    // Moves all of the elements of the `other` vector into the `v` vector.
+    fun verify_append<Element>(v: &mut vector<Element>, other: vector<Element>) {
+        spec {
+            /// Initialize the old vector `old_v` and `old_other` ghost variables
+            assume old_v<Element> == v;
+            assume old_other<Element> == other;
+        };
         Vector::reverse(&mut other);
-        while (!Vector::is_empty(&other)) Vector::push_back(lhs, Vector::pop_back(&mut other));
+        while ({
+            spec {
+                assert len(v) >= len(old_v<Element>);
+                assert len(other) <= len(old_other<Element>);
+                assert len(v) + len(other) == len(old_v<Element>) + len(old_other<Element>);
+                assert forall k in 0..len(old_v<Element>): v[k] == old_v<Element>[k];
+                assert forall k in 0..len(other): other[k] == old_other<Element>[len(old_other<Element>)-1-k];
+                assert forall k in len(old_v<Element>)..len(v): v[k] == old_other<Element>[k-len(old_v<Element>)];
+            };
+            !Vector::is_empty(&other)
+        }) {
+            Vector::push_back(v, Vector::pop_back(&mut other))
+        };
         Vector::destroy_empty(other);
     }
-    spec fun verify_append { // TODO: cannot verify loop
-        //ensures len(lhs) == old(len(lhs) + len(other)); //! A postcondition might not hold on this return path.
-        //ensures lhs[0..len(old(lhs))] == old(lhs); //! A postcondition might not hold on this return path.
-        //ensures lhs[len(old(lhs))..len(lhs)] == old(other); //! A postcondition might not hold on this return path.
+    spec fun verify_append {
+        ensures len(v) == old(len(v) + len(other)); //! A postcondition might not hold on this return path.
```

Comment:
Could you remove the comment which was the previous error message?

Reply:
Done.